deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+                 background_indexed: List[List[List[int]]],
+                 label,
+                 index: int=None):
+        super(IndexedMCTuplesWithBackgroundInstance, self).__init__(label, index)
+        self.answers_indexed = answers_indexed
+        self.background_indexed = background_indexed
+
+    @classmethod
+    @overrides
+    def empty_instance(cls):
+        return IndexedMCTuplesWithBackgroundInstance([], [], label=None, index=None)
+
+    @overrides
+    def get_lengths(self) -> Dict[str, int]:
+        # We care only about the longest slot here because all slots will be padded to the same length.
+        max_answer_length = 0
+        max_num_slots = 0
+        max_question_tuples = 0
+        for curr_answer in self.answers_indexed:
+            curr_max_num_slots, curr_max_answer_length = self.get_lengths_from_indexed_tuples(curr_answer)
+            max_num_slots = max(max_num_slots, curr_max_num_slots)
+            max_answer_length = max(max_answer_length, curr_max_answer_length)
+            max_question_tuples = max(max_question_tuples, len(curr_answer))
+        background_max_num_slots, background_max_slot_length = \
+            self.get_lengths_from_indexed_tuples(self.background_indexed)
+        max_num_slots = max(max_num_slots, background_max_num_slots)
+
+        return {'num_options': len(self.answers_indexed),
+                'num_question_tuples': max_question_tuples,
+                'num_background_tuples': len(self.background_indexed),
+                'question_slot_length': max_answer_length,
+                'background_slot_length': background_max_slot_length,
+                'num_slots': max_num_slots}
+
+    # Helper method for self.get_lengths() -- gets the max lengths for a list of indexed tuples
+    @staticmethod
+    def get_lengths_from_indexed_tuples(indexed_tuples: List[List[List[int]]]):
+        max_num_slots = 0
+        max_slot_length = 0
+        for indexed_tuple in indexed_tuples:
+            num_slots = len(indexed_tuple)
+            max_num_slots = max(max_num_slots, num_slots)
+            for slot in indexed_tuple:
+                max_slot_length = max(max_slot_length, len(slot))
+        return max_num_slots, max_slot_length
+
+    @overrides
+    def pad(self, max_lengths: Dict[str, int]):
+        """
+        Pads (or truncates) all slots in the answers_indexed and background_indexed to specified lengths.
+        Additionally, adjusts the number of slots in all tuples to the desired number: if there are fewer
+        slots than desired, will add empty lists at the end, and if there are more will
+        concatenate object slots.
+        Also, pads the number of answer options, the number of tuples per answer option, and the number
+        of background tuples to the specified lengths, adding empty ones as needed.
+        Note: Adjusting the number of slots happens before adjusting the lengths of all the slots.
```

Comment:
Can you explain this note a little more?  It's not clear from the comment that you're merging slots in some cases.